source/common/grpc/grpc_web_filter.cc

```diff
@@ -41,10 +41,10 @@ bool GrpcWebFilter::isGrpcWebRequest(const Http::HeaderMap& headers) {
 Http::FilterHeadersStatus GrpcWebFilter::decodeHeaders(Http::HeaderMap& headers, bool) {
   const Http::HeaderEntry* content_type = headers.ContentType();
   if (!isGrpcWebRequest(headers)) {
-    Http::Utility::sendLocalReply(*decoder_callbacks_, stream_destroyed_,
-                                  Http::Code::UnsupportedMediaType, EMPTY_STRING);
-    return Http::FilterHeadersStatus::StopIteration;
+    is_grpc_web_req_ = false;
```

Comment:
remove this line. as it has been initialized as false.

Reply:
I think so too but the tests won't pass because for some reason the true setting will persist. I thought tests will get instantiated one by one but I seem to be wrong. Any idea?

Reply:
Do you have the error log or stack trace?

Reply:
https://github.com/lyft/envoy/blob/master/test/common/grpc/grpc_web_filter_test.cc#L91

Here are it calls decodeHeaders in same filter instance multiple times. Please fix the test.

Reply:
Ping codesuki@. If you don't have time, I can pick over from here and fix the tests.

Reply:
@fengli79 IIRC @codesuki is on vacation so I think this will be fixed up early next week.